### PR TITLE
Fix `aws` bucket creation failure on `us-east-1`

### LINKED ISSUE #454: apply: api error InvalidLocationConstraint: The specified location-constraint is not valid
Same as https://github.com/boto/boto3/issues/125; happens when using `cloud = "aws"` and `region = "us-east"` (generic) or `region = "us-east-1"` (native equivalent).

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Closes #454; see the issue for context.

### CODE REVIEW

File: task/aws/resources/resource_bucket.go [hunk 1]
```diff
@@ -29,10 +29,12 @@ type Bucket struct {
...
 	createInput := s3.CreateBucketInput{
-		Bucket: aws.String(b.Identifier),
-		CreateBucketConfiguration: &types.CreateBucketConfiguration{
-			LocationConstraint: types.BucketLocationConstraint(b.Client.Region),
-		},
+		Bucket:                    aws.String(b.Identifier),
+		CreateBucketConfiguration: &types.CreateBucketConfiguration{},
+	}
+
+	if b.Client.Region != "us-east-1" {
```
Comment (casperdcl) [inline task/aws/resources/resource_bucket.go:36]: is future-proofing required? `us-east-*`?

Comment (0x2b3bfa0) [reply task/aws/resources/resource_bucket.go:36]: See https://github.com/boto/boto3/issues/125

**Review by casperdcl — APPROVED** [review]
:snowflake: + :technologist: = :snowman:

**Comment by 0x2b3bfa0** [comment]
☸️ + 🐈‍⬛ = ❄️

**Pull request merged**